x-pack/legacy/plugins/fleet/public/components/policy_delete_provider.tsx

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, useRef, useState } from 'react';
+import { EuiConfirmModal, EuiOverlayMask } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { ReturnTypeBulkDelete } from '../../common/return_types';
+import { useLibs, sendRequest } from '../hooks';
+
+interface Props {
+  children: (deletePolicies: deletePolicies) => React.ReactElement;
+}
+
+export type deletePolicies = (policies: string[], onSuccess?: OnSuccessCallback) => void;
+
+type OnSuccessCallback = (policiesUnenrolled: string[]) => void;
+
+export const PolicyDeleteProvider: React.FunctionComponent<Props> = ({ children }) => {
+  const libs = useLibs();
+  const [policies, setPolicies] = useState<string[]>([]);
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [isLoadingAgentsCount, setIsLoadingAgentsCount] = useState<boolean>(false);
+  const [agentsCount, setAgentsCount] = useState<number>(0);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const onSuccessCallback = useRef<OnSuccessCallback | null>(null);
+
+  const deletePoliciesPrompt: deletePolicies = (policiesToDelete, onSuccess = () => undefined) => {
+    if (
+      policiesToDelete === undefined ||
+      (Array.isArray(policiesToDelete) && policiesToDelete.length === 0)
+    ) {
+      throw new Error('No policies specified for deletion');
```

Comment:
This error is going to be user facing? if yes should we not translate it?

Reply:
this is an error that is most likely to be surfaced when calling `deletePoliciesPrompt` incorrectly, so it's not user facing 😊